code-challenges.test.js

```diff
@@ -14,43 +14,176 @@
+// _______________________________________________________________________________________________________________________________________
+
+
+// // --------------------2) Create a function that takes in an array and returns a new array of only odd numbers sorted from least to greatest.
+// Function name: oddOnlyArray
+// Function takes in an array and returns a new array of only odd numbers sorted from least to greatest 
+// Variables: fullArr1 and fullArr2
+// Use filter method to return an array with only odd numbers 
+// Use sort method to sort odd numbers from least to greatest 
+
+// // a) Create a test with expect statements for each of the variables provided.
+
+describe("oddOnlyArray", () => {
+    // Description of function
+    it("takes in an array and returns a new array of only odd numbers sorted from least to greatest", () => {
+        // const fullArr1 = [4, 9, 0, "7", 8, true, "hey", 7, 199, -9, false, "hola"]
+        // const fullArr2 = ["hello", 7, 23, -823, false, 78, null, "67", 6, "number"]
+        // const oddOnlyArray = (array) => {
+        //     let newArr = []
+        //     for(let i=0; i<array.length; i++){
+        //         if(array[i] % 2 !== 0){
+        //             newArr.push(array[i]).sort
+        //         }
+        // //         }
+        //         return newArr.sort((a, b) => a - b),newArr.filter(array => typeof array == 'number')
+        //     }
+        //Function: oddOnlyArray, Variable: fullArr1, expected output
+        expect(oddOnlyArray(fullArr1)).toEqual([-9, 7, 9, 199])
+        //Function: oddOnlyArray, Variable: fullArr2, expected output
+        expect(oddOnlyArray(fullArr2)).toEqual([-823, 7, 23])
+      })
+})
+
+
+//Fail received: 
+//oddOnlyArray › takes in an array and returns a new array of only odd numbers sorted from least to greatest
+ //   ReferenceError: oddOnlyArray is not defined
 
 const fullArr1 = [4, 9, 0, "7", 8, true, "hey", 7, 199, -9, false, "hola"]
 // Expected output: [-9, 7, 9, 199]
 
 const fullArr2 = ["hello", 7, 23, -823, false, 78, null, "67", 6, "number"]
 // Expected output: [-823, 7, 23]
 
-
-// b) Create the function that makes the test pass.
-
-
-// --------------------3) Create a function that takes in an array and returns an array of the accumulating sum. An empty array should return an empty array.
+// // b) Create the function that makes the test pass.
+//Function expression: oddOnlyArray
+     const oddOnlyArray = (array) => {
+            let newArr = []
+// Iterate over the length of the array
+            for(let i=0; i<array.length; i++){
+//Determine odd values
+                if(array[i] % 2 !== 0){
+//Sort odd values from least to greatest
+                    newArr.push(array[i]).sort
+                }
+            }
+//Use filter() method to filter out integers to avoid returning strings and booleans
+                return newArr.sort((a, b) => a - b),newArr.filter(array => typeof array == 'number')
+        }
```

Comment:
This approach certainly works.  Since you are already using filter in here, could this be simplified to not use a for loop?  Keep in mind, filter has a built-in conditional statement which you are already taking advantage of to find the typeof.  You could check if the typeof is "number" AND is odd as well.